apiserver/deployment.yaml.in

```diff
@@ -0,0 +1,138 @@
+apiVersion: apps/v1beta2
+kind: Deployment
+metadata:
+  name: spark-master
+spec:
+  selector:
+    matchLabels:
+      app: spark-master
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: spark-master
+    spec:
+      containers:
+        - name: spark-master
+          image: @spark_master_image@
+          imagePullPolicy: Always
+          ports:
+          - containerPort: 7077
+          - containerPort: 8080
+          volumeMounts:
+          - mountPath: /hail-vdc-sa-key
+            name: hail-vdc-sa-key
+          resources:
+            requests:
+              memory: "3.75G"
+              cpu: "0.9"
+      volumes:
+        - name: hail-vdc-sa-key
+          secret:
+            secretName: hail-vdc-sa-key
+---
+apiVersion: apps/v1beta2
+kind: Deployment
+metadata:
+  name: spark-worker
+spec:
+  selector:
+    matchLabels:
+      app: spark-worker
+  replicas: 2
+  template:
+    metadata:
+      labels:
+        app: spark-worker
+    spec:
+      tolerations:
+      - key: preemptible
+        value: "true"
+      containers:
+        - name: spark-worker
+          image: @spark_worker_image@
+          imagePullPolicy: Always
+          ports:
+          - containerPort: 8081
+          volumeMounts:
+          - mountPath: /hail-vdc-sa-key
+            name: hail-vdc-sa-key
+          resources:
+            requests:
+              memory: "7.5G"
+              cpu: "1.8"
+      volumes:
+        - name: hail-vdc-sa-key
+          secret:
+            secretName: hail-vdc-sa-key
+---
+apiVersion: apps/v1beta2
+kind: Deployment
+metadata:
+  name: apiserver
+  annotations:
+    hail-jupyter-image: @hail_jupyter_image@
+spec:
+  selector:
+    matchLabels:
+      app: apiserver
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: apiserver
+    spec:
+      containers:
+        - name: apiserver
+          image: @apiserver_image@
+          env:
+          - name: HAIL_APISERVER_SPARK_MASTER
+            value: spark://spark-master:7077
```

Comment:
why is this `spark://...`? Is it because the name of the service exposing port 7077 is named spark? Or is it from the makeFile where network=spark?

Reply:
From https://spark.apache.org/docs/latest/spark-standalone.html , isn't this how workers connect to the master?

```
Once started, the master will print out a spark://HOST:PORT URL for itself, which you can use to connect workers to it, or pass as the “master” argument to SparkContext. 
```